src/python/pants/jvm/target_types.py

```diff
@@ -68,6 +68,14 @@ class JvmJdkField(StringField):
     )
 
 
+class PrefixedJvmJdkField(JvmJdkField):
```

Comment:
Maybe `CodegenJvmJdkField`? Not a big deal, naming things is hard

Reply:
The fields technically aren't codegen-specific. "Prefixed" seems fine for now to me.